roles/ocp_on_libvirt/tasks/libvirt_host_up.yml

```diff
@@ -1,74 +1,124 @@
 ---
-- name: "List nodes for {{ host['name'] }}"
+- name: "List nodes"
   community.libvirt.virt:
-    name: "{{ host['name'] }}"
     command: list_vms
   register: _ool_vms_list
 
-- name: "Set ool_node_not_exists"
+- name: "Set ool_hosts_names"
+  ansible.builtin.set_fact:
+    ool_hosts_names: "{{ ool_hosts_names | default([]) + [host['name']]}}"
+  loop: "{{ hosts }}"
+  loop_control:
+    loop_var: host
+
+- name: "Set ool_nodes_list"
   ansible.builtin.set_fact:
-    ool_node_not_exists: "{{ host['name'] not in _ool_vms_list.list_vms }}"
+    ool_nodes_list: "{{ ool_hosts_names | difference(_ool_vms_list.list_vms) }}"
 
 - name: "Set ool_qemu_path"
   ansible.builtin.set_fact:
     ool_qemu_path: "{{ (ansible_distribution in ['CentOS', 'RedHat']) | ternary('/usr/libexec/', '/usr/bin/') }}"
 
 - name: "Copy image_src:"
   ansible.builtin.include_tasks: copy_image_src.yml
-  when: ool_node_not_exists
+  loop: "{{ hosts }}"
+  loop_control:
+    loop_var: host
+  when: host['name'] in ool_nodes_list
 
 - name: "Resize OS disk"
-  ansible.builtin.command: "qemu-img resize {{ libvirt_image_path }}/{{ host['name'] }}.qcow2 +{{ host['disk_size'] }}"
   become: "{{ libvirt_become }}"
-  when: host['disk_size'] is defined and ool_node_not_exists
+  ansible.builtin.command: "qemu-img resize {{ libvirt_image_path }}/{{ host['name'] }}.qcow2 +{{ host['disk_size'] }}"
+  loop: "{{ hosts }}"
+  loop_control:
+    loop_var: host
+  when: host['disk_size'] is defined and host['name'] in ool_nodes_list
```

Comment:
I feel that if `name` is going to be a required parameter there should be a validation at the top of the role, or force the end user to set it somehow e.g. make it a dictionary instead of a list